ui/components/app/signature-request/signature-request.js

```diff
@@ -0,0 +1,406 @@
+  Button,
+  Icon,
+  IconName,
+  Text,
+} from '../../component-library';
+import Box from '../../ui/box/box';
+///: END:ONLY_INCLUDE_IN
+
+import Message from './signature-request-message';
+import Footer from './signature-request-footer';
+
+const SignatureRequest = ({
+  txData,
+  signPersonalMessage,
+  signTypedMessage,
+  cancelPersonalMessage,
+  cancelTypedMessage,
+  signMessage,
+  cancelMessage,
+}) => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const trackEvent = useContext(MetaMetricsContext);
+  const t = useI18nContext();
+
+  const [hasScrolledMessage, setHasScrolledMessage] = useState(false);
+  const [showContractDetails, setShowContractDetails] = useState(false);
+  const [messageRootRef, setMessageRootRef] = useState(null);
+  const {
+    type,
+    msgParams: { from, data, origin, version },
+  } = txData;
+
+  const {
+    chainId,
+    type: providerName,
+    rpcPrefs,
+    nickname: providerNickName,
+  } = useSelector(getProviderConfig);
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const hardwareWalletRequiresConnection = useSelector((state) =>
+    doesAddressRequireLedgerHidConnection(state, from),
+  );
+  const isLedgerWallet = useSelector((state) => isAddressLedger(state, from));
+  const unconfirmedMessagesList = useSelector(unconfirmedMessagesHashSelector);
+  const unapprovedMessagesCount = useSelector(getTotalUnapprovedMessagesCount);
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const conversionRateSelected = useSelector(conversionRateSelector);
+  const conversionRate = useNativeCurrencyAsPrimaryCurrency
+    ? null
+    : conversionRateSelected;
+  const subjectMetadata = useSelector(getSubjectMetadata);
+  // not forwarded to component
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
+  const { address, balance, name } =
+    getAccountByAddress(allAccounts, from) || {};
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  // Used to show a warning if the signing account is not the selected account
+  // Largely relevant for contract wallet custodians
+  const selectedAccount = useSelector(getSelectedAccount);
+  ///: END:ONLY_INCLUDE_IN
+
+  let cancel;
+  let sign;
+
+  if (type === MESSAGE_TYPE.PERSONAL_SIGN) {
+    cancel = cancelPersonalMessage;
+    sign = signPersonalMessage;
+  } else if (type === MESSAGE_TYPE.ETH_SIGN_TYPED_DATA) {
+    cancel = cancelTypedMessage;
+    sign = signTypedMessage;
+  } else if (type === MESSAGE_TYPE.ETH_SIGN) {
+    cancel = cancelMessage;
+    sign = signMessage;
+  }
+
+  const memoizedParseMessage = memoize(() => {
+    const { message, domain = {}, primaryType, types } = JSON.parse(data);
+    const sanitizedMessage = sanitizeMessage(message, primaryType, types);
+    return { sanitizedMessage, domain, primaryType };
+  });
```

Comment:
Above does not have a dependency ? May not be use case for memoize.

Reply:
@NiranjanaBinoy : please check above

Reply:
@jpuri I have added this back based on [this comment](https://github.com/MetaMask/metamask-extension/pull/19104#discussion_r1255690532)